app.py

```diff
@@ -0,0 +1,600 @@
+                            ))
+                    }
+                })
+
+        record.update({'engine': engine_dict})
+
+        # Adds Running Gear collection to record
+        running_dict = []
+        for item in running:
+            product = request.form.get('running_'+item["part_id"]+'_product')
+            if product is not None:
+                running_dict.append({
+                    item["part_id"]: {
+                        'product': product,
+                        'link': request.form.get(
+                                'running_'+item["part_id"]+'_link'
+                            ),
+                        'price': float(request.form.get(
+                                'running_'+item["part_id"]+'_price'
+                            ))
+                    }
+                })
+
+        record.update({'running': running_dict})
+
+        # Adds Interior collection to record
+        interior_dict = []
+        for item in interior:
+            product = request.form.get('interior_'+item["part_id"]+'_product')
+            if product is not None:
+                interior_dict.append({
+                    item["part_id"]: {
+                        'product': product,
+                        'link': request.form.get(
+                                'interior_'+item["part_id"]+'_link'
+                            ),
+                        'price': float(request.form.get(
+                                'interior_'+item["part_id"]+'_price'
+                            ))
+                    }
+                })
+
+        record.update({'interior': interior_dict})
+
+        # Update in Mongo
+        builds.update({"_id": ObjectId(build_id)}, {'$set': record})
+        flash('Build Updated', category='warning')
+        return redirect(url_for('view_record', build_id=build_id))
+
+    build = mongo.db.builds.find_one({
+        "_id": ObjectId(build_id)
+    })
+
+    exterior = mongo.db.exterior.find()
+    engine = mongo.db.engine.find()
+    running = mongo.db.runninggear.find()
+    interior = mongo.db.interior.find()
+
+    return render_template("edit.html", build=build, exterior=list(exterior),
+                           engine=list(engine), running=list(running),
+                           interior=list(interior))
+
+
+# Delete a record
+@app.route("/build/<build_id>/delete")
+@login_required
+def delete_record(build_id):
+    mongo.db.builds.remove({
+        '_id': ObjectId(build_id)
+    })
+    flash('Build Deleted', category='danger')
+    return redirect(url_for('builds'))
+
+
+# Votes
+@app.route('/build/like/<build_id>', methods=['POST'])
+def like_build(build_id):
+    build = mongo.db.builds
+
+    current_build = mongo.db.builds.find_one({
+        "_id": ObjectId(build_id)
+    })
+
+    array_length = int(len(current_build['votes']['like']['users_liked']))
+
+    liked_by = []
+
+    for x in range(0, array_length):
```

Comment:
instead of doing this loop, do...
liked_by = list(current_build['votes']['like']['users_liked'])
array_length = len(users_liked)
Use better variable names tho..